### PR TITLE
Missing tiles in debug.log separate from missing with looks_like

### PR DESCRIPTION
**Pull request by johnstoecker**
#### Summary: 

`Summary: None`

#### Purpose of change
Currently, when reloading a tileset, debug.log outputs missing tiles, even if their "looks_like" definition is in the tileset.

For example, when the RetroDays tileset is reloaded, debug.log displays "alexandrite_silver_ring" as missing. However, "alexandrite_silver_ring" has copy_from "garnet_silver_ring" which looks_like "silver_ring". The tile_config.json in RetroDays has "silver_ring" defined, so we might want to separately log it.

Cleaning up this output will help tileset maintainers more easily find missing tiles.

#### Describe the solution
Use the find_tile_looks_like, and if that is defined in the tileset, don't output as missing.

The output would look like:

```
Missing item: kevlar_silver_ring
Missing item (but looks_like tile exists): alexandrite_silver_ring
```

#### Testing
Reloaded a few tilesets, made sure that they worked in game. Checked the output in debug.log for missing tiles. The missing ones are actually missing, and the ones with missing looks_like still appear, but with their generic tile.

### CODE REVIEW

**Comment by ZhilkinSerg** [comment]
Tiles for entities with looks_like are still missing from the tileset and should still be reported. Maybe in !a?a separate line in the report.

**Comment by johnstoecker** [comment]
Ah, I see -- my motivation was originally to update a tileset. Then, when I looked at the number of missing tiles, it seemed very daunting....

Is the intention of "looks_like" to indicate that something should look exactly like something else (i.e. alexandrite_silver_ring vs garnet_silver_ring), or is that something that we should leave up to the tileset maintainers?

**Comment by ZhilkinSerg** [comment]
`looks_like` is just a fallback mechanism - it is used when tileset is missing a tile for an object. We are adding it to objects, but that does not necessary means we don't have to add unique tiles for that objects anymore.

**Comment by anothersimulacrum** [comment]
Yes, the proper way to say that we want tiles to look the same is to specify them within the tileset definition.

**Comment by johnstoecker** [comment]
Updated to show missing looks_like in a new line. With that in mind, my proposed change might make things worse, and not better -- tile maintainers would be more likely to rely on looks_like, and not the proper way of copying ids inside the tile_config, or creating new tiles for each looks_like.

I'll leave the PR open, but feel to close it.

**Comment by ZhilkinSerg** [comment]
I don't see how it can make things worse - you've basically just split output to two rows. Number of reported tiles would be the same.

File: src/cata_tiles.cpp [hunk 1]
```diff
@@ -3532,54 +3532,56 @@ point cata_tiles::player_to_screen( const point &p ) const
...
     std::string missing_list;
+    std::string missing_with_looks_like_list;
     for( ; begin != end; ++begin ) {
         const std::string id_string = id_func( begin );
-        if( !tileset_ptr->find_tile_type( prefix + id_string ) ) {
-            missing++;
+
+        std::string mutable_id_string = id_string;
+
+        if( !tileset_ptr->find_tile_type( prefix + id_string ) && !find_tile_looks_like( mutable_id_string, category) ) {
```
Comment (kevingranade) [inline src/cata_tiles.cpp:3545]:
```suggestion
        if( !tileset_ptr->find_tile_type( prefix + id_string ) &&
            !find_tile_looks_like( mutable_id_string, category) ) {
```

File: src/cata_tiles.cpp [hunk 2]
```diff
@@ -3532,54 +3532,56 @@ point cata_tiles::player_to_screen( const point &p ) const
...
             missing_list.append( id_string + " " );
-        } else {
-            present++;
+        } else if( !tileset_ptr->find_tile_type( prefix + id_string ) ) {
+            missing_with_looks_like_list.append( id_string + " " );
         }
     }
-    DebugLog( D_INFO, DC_ALL ) << "Missing " << label << ": " << missing_list;
+    DebugLog( D_INFO, DC_ALL ) << "Missing " << TILE_CATEGORY_IDS[category] << ": " << missing_list;
+    DebugLog( D_INFO, DC_ALL ) << "Missing " << TILE_CATEGORY_IDS[category] << " (but looks_like tile exists): " << missing_with_looks_like_list;
```
Comment (kevingranade) [inline src/cata_tiles.cpp:3552]:
```suggestion
    DebugLog( D_INFO, DC_ALL ) << "Missing " << TILE_CATEGORY_IDS[category] <<
        " (but looks_like tile exists): " << missing_with_looks_like_list;
```

File: src/cata_tiles.cpp [hunk 3]
```diff
@@ -3532,54 +3532,58 @@ point cata_tiles::player_to_screen( const point &p ) const
...
+    std::string missing_with_looks_like_list;
     for( ; begin != end; ++begin ) {
         const std::string id_string = id_func( begin );
-        if( !tileset_ptr->find_tile_type( prefix + id_string ) ) {
-            missing++;
+
+        std::string mutable_id_string = id_string;
+
+        if( !tileset_ptr->find_tile_type( prefix + id_string ) &&
+            !find_tile_looks_like( mutable_id_string, category) ) {
```
Comment (kevingranade) [inline src/cata_tiles.cpp:3546]:
```suggestion
            !find_tile_looks_like( mutable_id_string, category ) ) {
```

File: src/cata_tiles.cpp [hunk 4]
```diff
@@ -3532,54 +3532,58 @@ point cata_tiles::player_to_screen( const point &p ) const
...
-        } else {
-            present++;
+        } else if( !tileset_ptr->find_tile_type( prefix + id_string ) ) {
+            missing_with_looks_like_list.append( id_string + " " );
         }
     }
-    DebugLog( D_INFO, DC_ALL ) << "Missing " << label << ": " << missing_list;
+    DebugLog( D_INFO, DC_ALL ) << "Missing " << TILE_CATEGORY_IDS[category] << ": " << missing_list;
+    DebugLog( D_INFO, DC_ALL ) << "Missing " << TILE_CATEGORY_IDS[category] <<
+        " (but looks_like tile exists): " << missing_with_looks_like_list;
```
Comment (kevingranade) [inline src/cata_tiles.cpp:3554]:
```suggestion
    DebugLog( D_INFO, DC_ALL ) << "Missing " << TILE_CATEGORY_IDS[category] <<
                               " (but looks_like tile exists): " << missing_with_looks_like_list;
```

**Pull request merged**